### PR TITLE
Deprecate UserForcingFunction

### LINKED ISSUE #9876: BodyForce and UserForcingFunction are nearly the same.
### Description of the enhancement or error report
Code duplication among Kernels.

### Rationale for the enhancement or information for reproducing the error
The BodyForce and UserForcingFunction serve the same purpose. UserForcingFunction should probably go away in favor of BodyForce, which is slightly more generic. 

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)
UserForcingFunction can be deprecated, so there should be minimal impact.

### PR DESCRIPTION
**Pull request by aeslaughter**
(closes #9876)

### CODE REVIEW

**Comment by aeslaughter** [comment]
This is ready.

File: docs/content/documentation/systems/Distributions/stochastic_tools/LognormalDistribution.md [hunk 1]
```diff
@@ -1,29 +0,0 @@
-# LognormalDistribution
```
Comment (permcody) [inline docs/content/documentation/systems/Distributions/stochastic_tools/LognormalDistribution.md]: You probably don't want to delete this!

File: framework/src/base/Factory.C [hunk 2]
```diff
@@ -84,77 +84,25 @@ Factory::restrictRegisterableObjects(const std::vector<std::string> & names)
...
-}
-
 void
 Factory::deprecatedMessage(const std::string obj_name)
 {
-  std::map<std::string, time_t>::iterator time_it = _deprecated_time.find(obj_name);
-
   // If the object is not deprecated return
-  if (time_it == _deprecated_time.end())
+  std::set<std::string>::iterator iter = _deprecated.find(obj_name);
```
Comment (permcody) [inline framework/src/base/Factory.C]: `auto`

File: framework/src/base/Factory.C [hunk 3]
```diff
@@ -84,77 +84,25 @@ Factory::restrictRegisterableObjects(const std::vector<std::string> & names)
...
-
-    // Append replacement object, if it exsits
-    if (name_it != _deprecated_name.end())
-      msg << "Update your application using the '" << name_it->second << "' object";
-
-    // Produce the error message
-    mooseError(msg.str());
-  }
+  // Append replacement object, if it exsits
+  std::map<std::string, std::string>::iterator name_it = _deprecated_with_replace.find(obj_name);
```
Comment (permcody) [inline framework/src/base/Factory.C]: `auto`

File: test/tests/misc/block_boundary_material_check/dgkernel_check_block.i [hunk 4]
```diff
@@ -1,31 +0,0 @@
-[Mesh]
```
Comment (permcody) [inline test/tests/misc/block_boundary_material_check/dgkernel_check_block.i]: Did you mean to delete this?

File: test/tests/misc/block_boundary_material_check/dgkernel_check_boundary.i [hunk 5]
```diff
@@ -1,31 +0,0 @@
-[Mesh]
```
Comment (permcody) [inline test/tests/misc/block_boundary_material_check/dgkernel_check_boundary.i]: Did you mean to delete this?

File: test/tests/misc/block_boundary_material_check/tests [hunk 6]
```diff
@@ -1,22 +0,0 @@
-[Tests]
```
Comment (permcody) [inline test/tests/misc/block_boundary_material_check/tests]: Did you mean to delete this?

File: test/tests/vectorpostprocessors/csv_reader/transfer/master.i [hunk 7]
```diff
@@ -1,71 +0,0 @@
-[Mesh]
```
Comment (permcody) [inline test/tests/vectorpostprocessors/csv_reader/transfer/master.i]: Did you mean to delete this?

File: test/tests/vectorpostprocessors/csv_reader/transfer/sub.i [hunk 8]
```diff
@@ -1,49 +0,0 @@
-[Mesh]
```
Comment (permcody) [inline test/tests/vectorpostprocessors/csv_reader/transfer/sub.i]: Did you mean to delete this?

File: test/tests/vectorpostprocessors/csv_reader/transfer/tests [hunk 9]
```diff
@@ -1,10 +0,0 @@
-[Tests]
```
Comment (permcody) [inline test/tests/vectorpostprocessors/csv_reader/transfer/tests]: Did you mean to delete this?

**Review by permcody — CHANGES_REQUESTED** [review]

**Review by permcody — APPROVED** [review]

**Pull request merged**